### PR TITLE
Add start/stop auxiliary-heating and verify_spin support

### PR DESCRIPTION
**Pull request by prvakt**
- Added auxiliary heater support (start/stop). For some reason progress is not published via MQTT so request is just send and not awaited for COMPLETED_SUCCESS status
- Added verify_spin method which returns `CORRECT_SPIN` or `INCORRECT_SPIN` and S-Pin status

### CODE REVIEW

File: myskoda/models/spin.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+@dataclass
+class SpinStatus(DataClassORJSONMixin):
+    state: str
+    remainingTries: int
+    lockedWaitingTimeInSeconds: int
+
+@dataclass
+class Spin(DataClassORJSONMixin):
+    verificationStatus: VerificationStatus
+    spinStatus: SpinStatus | None = field(default=None)
```
Comment (dvx76) [inline myskoda/models/spin.py:25]: Please use `snake_case` for class attributes and and alias in `field_options` to match the field name in the API response. E.g.

```suggestion
@dataclass
class SpinStatus(DataClassORJSONMixin):
    state: str
    remaining_tries: int = field(metadata=field_options(alias="remainingTries"))
    locked_waiting_time_in_seconds: int = field(metadata=field_options(alias="lockedWaitingTimeInSeconds"))

@dataclass
class Spin(DataClassORJSONMixin):
    verification_status: VerificationStatus = field(metadata=field_options(alias="verificationStatus"))
    spin_status: SpinStatus | None = field(default=None, metadata=field_options(alias="verificationSspinStatustatus"))
```

File: myskoda/models/spin.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
+"""Models for responses of api/v2/garage/vehicles/{vin}."""
+
+import logging
```
Comment (WebSpider) [inline myskoda/models/spin.py:3]: nit: Models do not need logging

File: myskoda/rest_api.py [hunk 3]
```diff
@@ -267,6 +281,32 @@ async def start_air_conditioning(self, vin: str, temperature: float) -> None:
             json=json_data,
         )
 
+    async def stop_auxiliary_heating(self, vin: str) -> None:
+        """Stop the auxiliary heating."""
+        _LOGGER.debug("Stopping auxiliary heating for vehicle %s", vin)
+        await self._make_post_request(url=f"/v2/air-conditioning/{vin}/auxiliary-heating/stop")
+
+    async def start_auxiliary_heating(self, vin: str, temperature: float, spin: str = "") -> None:
+        """Start the auxiliary heating."""
```
Comment (WebSpider) [inline myskoda/rest_api.py:290]: Will the request actually work with an empty-string S-PIN?
If not, please remove the default.

File: myskoda/rest_api.py [hunk 4]
```diff
@@ -267,6 +281,32 @@ async def start_air_conditioning(self, vin: str, temperature: float) -> None:
...
+            "Starting auxiliary heating for vehicle %s with temperature %s",
+            vin,
+            str(temperature),
+        )
+        json_data = {
+            "heaterSource": "AUTOMATIC",
+            "airConditioningWithoutExternalPower": True,
+            "spin": spin,
+            "targetTemperature": {
+                "temperatureValue": str(temperature),
```
Comment (WebSpider) [inline myskoda/rest_api.py:301]: You seem to be forcing `float` conversion to `string` here. Will that work for all accepted inputs?
eg: can we set the temperature to `20.005` ?

File: myskoda/models/spin.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
+"""Models for responses of api/v2/garage/vehicles/{vin}."""
```
Comment (WebSpider) [inline myskoda/models/spin.py:1]: nit: This should be `/v2/air-conditioning/{vin}/auxiliary-heating/stop`

**Review by WebSpider — CHANGES_REQUESTED** [review]
Thanks for your PR, much appreciated!

Looks good overall, some minor remarks.

File: myskoda/rest_api.py (hunk 4 above)
Comment (prvakt) [reply myskoda/rest_api.py]: implemented rounding to 1 decimal with 0,5 step and updated all methods

**Review by dvx76 — DISMISSED** [review]
LGTM

**Comment by prvakt** [comment]
please wait with approval, planning to test something tomorrow because maybe we are using wrong endpoint for the auxiliary-heating MQTT subscription (https://github.com/skodaconnect/myskoda/issues/179)

**Comment by WebSpider** [comment]
Converted to draft PR, so it won't be merged by accident.

**Comment by prvakt** [comment]
Good news, after changing mqtt endpoint for auxiliary-heater as suggested in (https://github.com/skodaconnect/myskoda/issues/179), I can receive state updates. 
This PR is finished and can be merged ;)

File: myskoda/const.py [hunk 6]
```diff
@@ -37,7 +37,7 @@
 MQTT_SERVICE_EVENT_TOPICS = [
     "air-conditioning",
     "charging",
-    "departure"
+    "departure",
```
Comment (WebSpider) [inline myskoda/const.py:40]: Is this new topic needed for aux heating/verify spin or is it something you discovered along the way?

Comment (prvakt) [reply myskoda/const.py:40]: `departure` and `auxiliary-heating/start-stop-auxiliary-heating` was found by @fursov in #179  so I have added it too

Comment (WebSpider) [reply myskoda/const.py:40]: Good, thanks for your contribution!

**Review by WebSpider — APPROVED** [review]
Lgtm

**Review by dvx76 — APPROVED** [review]

**Pull request merged**